### PR TITLE
UPSTREAM: 31818: suggest use of `oc types` in oc get output

### PR DESCRIPTION
**Pull request by juanvallejo**
UPSTREAM: https://github.com/kubernetes/kubernetes/pull/31818

Related Trello card:
https://trello.com/c/04lpfTQR/451-8-cli-improved-flows-and-ux-in-the-cli-evg-ux-p3

This patch improves usability flow, making it easier for a user to
discover the command `oc types` through `oc get`. While `oc types`
provides both resource names and descriptions, `oc get` only provides a
list of resources, and no immediate way to find out more about them.

cc @fabianofranz

### CODE REVIEW

**Comment by juanvallejo** [comment]
[test]

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go [hunk 1]
```diff
@@ -160,7 +172,8 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 
 	if len(args) == 0 && len(options.Filenames) == 0 {
 		fmt.Fprint(out, "You must specify the type of resource to get. ", valid_resources)
-		return cmdutil.UsageError(cmd, "Required resource not specified.")
+		usageString := fmt.Sprintf("Required resource not specified.\nUse \"%s types\" for a description of each resource.", options.FullCmdName)
```
Comment (kargakis) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go]: How about proposing `oc explain`? We need to deprecate `oc types` in favor of the former.

Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go]: That sounds like a better idea. I'll go ahead and update this

**Comment by juanvallejo** [comment]
re[test]

**Comment by juanvallejo** [comment]
re[test]

**Comment by fabianofranz** [comment]
LGTM, waiting for upstream and post-1.3.

**Comment by fabianofranz** [comment]
Needs rebase.

**Comment by juanvallejo** [comment]
@fabianofranz PTAL

**Comment by fabianofranz** [comment]
Waiting for feedback upstream.

**Comment by juanvallejo** [comment]
[conformance](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/6419/) test flaked on https://github.com/openshift/origin/issues/11079 re[test]

**Comment by fabianofranz** [comment]
@juanvallejo is this already up-to-date with the upstream version?

**Comment by juanvallejo** [comment]
@fabianofranz 
>  is this already up-to-date with the upstream version?

Yes, made sure to keep this one in sync from the feedback I received upstream :)

**Comment by fabianofranz** [comment]
@juanvallejo is the swagger spec commit still needed?

**Comment by juanvallejo** [comment]
@fabianofranz I don't think so, I'll go ahead and fix this

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**